### PR TITLE
Get rid of temporary properties file

### LINKED ISSUE #201: JDBCBackend.__del__ is noisy on Windows
@GamzeUnlu95 reports the following:
```
Exception ignored in: <function JDBCBackend.__del__ at 0x0000028D9F015400>
Traceback (most recent call last):
 File "c:\users\unlu\documents\github\ixmp\ixmp\backend\jdbc.py", line 210, in __del__
   Path(self._properties_file).unlink()
 File "C:\Users\unlu\AppData\Local\Continuum\anaconda3\lib\pathlib.py", line 1287, in unlink
   self._accessor.unlink(self)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\unlu\\AppData\\Local\\Temp\\tmp_s2qwnou.properties'
```

This is because the destructor (__del__) of JDBCBackend is trying to remove the temporary properties file: https://github.com/iiasa/ixmp/blob/952d1345c7df6e8b79ce64e4333847b39f7bfb79/ixmp/backend/jdbc.py#L208-L212

The solution is probably to call .close_db() before removing the file.

### PR DESCRIPTION
**Pull request by zikolach**
Closes #201.

- parameterize platform constructor with Properties class instance

### CODE REVIEW

**Comment by khaeru** [comment]
@zikolach please do hit the "Ready for review" button to take this out of "Draft" state if that is indeed the case.

File: ixmp/backend/jdbc.py [hunk 1]
```diff
@@ -174,7 +179,9 @@ def __init__(self, jvmargs=None, **kwargs):
             dbprops = Path(kwargs.pop('dbprops'))
             if dbprops.exists() and dbprops.is_file():
                 # Existing properties file
-                properties_file, info = _read_dbprops(dbprops)
+                properties = _read_properties(dbprops)
+                if 'jdbc.url' not in properties:
```
Comment (zikolach) [inline ixmp/backend/jdbc.py]: @khaeru can you check whether this if can work?

Comment (khaeru) [reply ixmp/backend/jdbc.py]: Added 293d855 anyway.

**Comment by zikolach** [comment]
@khaeru please feel free to merge/squash changes

**Comment by khaeru** [comment]
@zikolach immediately after merging I realized this PR was not added to RELEASE NOTES. Please use the PR template in the future.

**Comment by zikolach** [comment]
> use the PR template

That's true. Sorry for that. As you saw my changes were quite unpolished and I considered them as a draft version. Should we restore branch and re-merge adding release notes?

**Comment by khaeru** [comment]
It's okay, not urgent.

**Pull request merged**